src/openvic-simulation/economy/Producer.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <map>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+using money_t = OpenVic::fixed_point_t;
+using good_quantity_t = OpenVic::fixed_point_t;
+
+namespace OpenVic {
+	class Producer {
+	public:
+		ProductionType PROPERTY(production_type)
```

Comment:
```suggestion
		ProductionType const& PROPERTY(production_type);
```
We won't store copies of `ProductionType`s, rather just a reference to the global copy of them created when loading the defines (reference rather than pointer, because `production_type` should never be null).